src/inabox/inabox-viewport.js

```diff
@@ -157,14 +224,109 @@ export class ViewportBindingInabox {
     }
   }
 
+  /** @override */
+  updateLightboxMode(lightboxMode) {
+    if (lightboxMode) {
+      return this.tryToEnterOverlayMode_();
+    }
+    return this.leaveOverlayMode_();
+  }
+
+  /**
+   * @return {!Promise}
+   * @private
+   */
+  tryToEnterOverlayMode_() {
+    // TODO(alanorozco): Update viewport measurement from host message.
+    return this.prepareFixedContainer_()
+        .then(() => this.requestFullOverlayFrame_());
+  }
+
+  /**
+   * @return {!Promise}
+   * @private
+   */
+  leaveOverlayMode_() {
+    return this.requestResetFullOverlayFrame_()
+        .then(() => this.resetFixedContainer_());
+  }
+
+  /**
+   * Prepares the "fixed" container before expanding frame.
+   * @return {!Promise}
+   * @private
+   */
+  prepareFixedContainer_() {
+    const fixedContainer = this.getFixedContainer_();
+
+    if (!fixedContainer) {
+      dev().warn(TAG, 'No fixed container inside frame, content will shift.');
+      return Promise.resolve();
+    }
+
+    return prepareFixedContainer(this.win, dev().assertElement(fixedContainer));
+  }
+
+  /**
+   * Resets the "fixed" container to its original position after collapse.
+   * @return {!Promise}
+   * @private
+   */
+  resetFixedContainer_() {
+    const fixedContainer = this.getFixedContainer_();
+
+    if (!fixedContainer) {
+      dev().warn(TAG, 'No fixed container inside frame, content will shift.');
+      return Promise.resolve();
+    }
+
+    return resetFixedContainer(this.win, dev().assertElement(fixedContainer));
+  }
+
+  /**
+   * @return {!Promise}
+   * @private
+   */
+  requestFullOverlayFrame_() {
+    return new Promise((resolve, reject) => {
+      this.iframeClient_.makeRequest(
+          MessageType.FULL_OVERLAY_FRAME,
```

Comment:
Would that be a problem? A sequence of the same call should have the same result anyways, so there's no need to map responses on the receiving end.